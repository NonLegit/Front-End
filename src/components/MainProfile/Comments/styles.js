import { Box } from '@mui/material';
import { styled } from '@mui/material/styles';

export const PostsQueueBox = styled(Box)(() => ({
  display: 'flex',
  flexDirection: 'column',
  backgroundColor: 'white',
  borderRadius: 3,
  marginBottom: 5,
  position: 'relative',
}));

export const CommentText = styled(Box)(({ theme, coloring, hover }) => ({
  marginRight: 5,
  color: coloring,
  fontSize: 13,
  ...((coloring === undefined) && {
    color: theme.palette.primary.main,
  }),
  ...((hover === 'true') && {
    '&:hover': {
      textDecoration: 'underline',
      cursor: 'pointer',
    },
  }),

}));
