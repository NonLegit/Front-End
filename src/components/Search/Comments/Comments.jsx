import {
  Avatar, Box, Typography,
} from '@mui/material';
import calculateTime from '../../../utils/calculateTime';
import numFormatter from '../../../utils/MembersNum';
import {
  PostContainer, PostInfo, PostInfoLink, PostTitle,
  PostRich,
  CommentContainer,
  CommentReach,
  GoToThread,
  PostLink,
} from './style';

/**
 * Search by Comments entity
 * @component
 * @return {React.Component} - Search by community entity
 */
function Comments(props) {
  const { Comment } = props;
  return (
    <PostContainer sx={{ '&:hover': { border: 1 } }}>
      <Box width="100%">
        <Box sx={{ padding: '16px 16px 4px' }}>
          <PostInfo>
            <Avatar
              src={Comment.post.owner.icon}
              sx={{
                width: 20,
                height: 20,
              }}
              alt="Profile Image"
            />
            <PostInfoLink to={`/r/${Comment.post.owner.fixedName}`} color="#000" fontWeight="bolder">
              r/
              {Comment.post.owner.name}
            </PostInfoLink>
            <Box color="#787C7E" fontWeight={300} display="flex" gap="4px" flexWrap="wrap">
              <span>
                •
              </span>
              <div>Posted By</div>
              <Avatar
                src={Comment.post.author.profilePicture}
                sx={{
                  width: 20,
                  height: 20,
                }}
                alt="Profile Image"
              />
              <PostInfoLink to={`/user/${Comment.post.author.userName}`} color="inherit" fontWeight="normal">
                u/
                {Comment.post.author.displayName}
                {' '}
                {calculateTime(Comment.post.createdAt)}
              </PostInfoLink>
            </Box>
          </PostInfo>
        </Box>

        <PostTitle>
          <PostLink to={`../subreddit/${Comment.post?.owner?.fixedName}/comments/${Comment.post?._id}`}>
            <Typography
              variant="h6"
              component="h3"
              sx={{
                fontSize: 12, marginBottom: '12px', fontWeight: '400',
              }}
            >
              {Comment.post.title.replace(/<[^>]+>/g, '')}
            </Typography>
          </PostLink>
          <CommentContainer to={`../subreddit/${Comment.post?.owner?.fixedName}/comments/${Comment.post?._id}/${Comment?._id}`}>
            <Avatar
              src={Comment.author.profilePicture}
              sx={{
                width: 20,
                height: 20,
                margin: 1,
              }}
              alt="Profile Image"
            />
            <Box>
              <PostInfo py={1}>

                <PostInfoLink to={`/user/${Comment.author.userName}`} color="#000" fontWeight="bolder">
                  {Comment.author.displayName}
                </PostInfoLink>
                <Box color="#787C7E" fontWeight={300} display="flex" gap="4px" flexWrap="wrap">
                  <Typography component="span" mx="4px" sx={{ fontSize: '6px', display: 'flex', alignItems: 'center' }}>
                    •
                  </Typography>
                  <PostInfoLink to="/" color="inherit" fontWeight="normal">
                    {calculateTime(Comment.createdAt)}
                  </PostInfoLink>
                </Box>
              </PostInfo>
              {/* question */}
              <Typography variant="h6" component="h3" sx={{ fontSize: 14, marginBottom: '12px', fontWeight: '400' }}>
                {Comment.text.replace(/<[^>]+>/g, '')}
              </Typography>
              <CommentReach>
                <Typography variant="span" sx={{ marginRight: 3, padding: 0 }}>
                  {numFormatter(Comment.votes)}
                  {' '}
                  upvotes
                  {' '}
                </Typography>
                <Typography variant="span" sx={{ marginRight: 3, padding: 0 }}>
                  {numFormatter(Comment.Comment)}
                  {' '}
                  awards
                  {' '}
                </Typography>
              </CommentReach>
            </Box>
          </CommentContainer>
        </PostTitle>
        <GoToThread to={`../subreddit/${Comment.post?.owner?.fixedName}/comments/${Comment.post?._id}/${Comment?._id}`}>
          Go to thread
        </GoToThread>
        <PostRich mt={5}>
          <Typography variant="span" sx={{ marginRight: 3 }}>
            {numFormatter(Comment.post.votes)}
            {' '}
            upvotes
            {' '}
          </Typography>
          <Typography variant="span" sx={{ marginRight: 3 }}>
            {numFormatter(Comment.post.commentCount)}
            {' '}
            comments
            {' '}
          </Typography>
        </PostRich>
      </Box>
    </PostContainer>
  );
}

export default Comments;
