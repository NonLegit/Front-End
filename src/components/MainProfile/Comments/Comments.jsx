import { Divider } from '@mui/material';
import { useParams } from 'react-router-dom';
import { useEffect, useContext, useState } from 'react';
import { PostsQueueBox } from './styles';
import CommentsContent from './CommentsContent/CommentsContent';
import CommentsHeader from './CommentsHeader/CommentsHeader';
import { CommunitiesContext } from '../../../contexts/CommunitiesModeratorContext';

/**
 * Comments component shows all comments over a post
 *
 * @component Comments
 * @property {object} entity - conatin all info about the post and the list of comments needed to be shown
 * @property {boolean} profile - a signal indecates whether i was in the profile or not
 * @property {boolean} noheader - a signal ensure to remove the header of the post if the post is already shown
 * @property {boolean} overview - a signal indecates whether the comment was in overview page or not
 * @returns {React.Component} Comments
 */

function Comments(props) {
  const {
    entity, profile, noheader, overview,
  } = props;
  const { username } = useParams();
  const { communities } = useContext(CommunitiesContext);
  const [modList, setModList] = useState(false);

  /**
   * the useState below used to ensure whether the comment was in a subreddit i moderator in or not
   */

  useEffect(() => {
    console.log(entity);
    if (entity?.ownerType === 'Subreddit' && communities?.filter((element) => element.fixedName === entity?.owner.name).length > 0) {
      setModList(true);
    }
  }, [communities]);

  return (
    <PostsQueueBox>
      <CommentsHeader
        post={entity}
        noheader={noheader}
      />
      <Divider variant="middle" />
      { entity?.comments?.map((comment, index) => (
        <>
          <CommentsContent
            key={`${index + 0}`}
            postid={entity?._id}
            comment={comment}
            op={entity?.author?.name === username}
            modList={modList}
            profile={profile}
            overview={overview}
            ownerType={entity?.ownerType}
            owner={entity?.owner.name}
          />
          <Divider variant="middle" />
        </>
      ))}

    </PostsQueueBox>

  );
}

export default Comments;
