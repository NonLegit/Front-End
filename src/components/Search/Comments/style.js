import { Box } from '@mui/material';
import { styled } from '@mui/material/styles';
import { Link } from 'react-router-dom';

export const PostContainer = styled(Box)(() => ({
  width: '100%',
  backgroundColor: '#fff',
  display: 'flex',
  borderRadius: 3,
  border: '1px solid #ccc',
  alignItems: 'center',
  '&:hover':
  {
    borderColor: '#999',
  },
  margin: 0,
  cursor: 'pointer',
}));

export const PostInfo = styled(Box)(() => ({
  display: 'flex',
  fontSize: '12px',
  gap: '4px',
}));

export const PostInfoLink = styled(Link)(({ color, fontWeight }) => ({
  fontWeight,
  color,
  textDecoration: 'none',
  '&:hover': {
    textDecoration: 'underline',
  },
}));

export const PostTitle = styled(Link)(() => ({
  textDecoration: 'none',
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'spacebetween',
  padding: '0 16px',
  color: '#000',
  '& .MuiTypography-root': {
    lineHeight: 1.2,
    fontWeight: 500,
  },
  '&:hover': {
    textDecoration: 'none',
  },
}));

export const PostMedia = styled(Box)(() => ({
  display: 'flex',
  flexDirection: 'column',
  width: 138,
  height: 98,
  justifyContent: 'center',
}));

export const PostRich = styled('div')({
  paddingRight: 16,
  paddingLeft: 16,
  paddingTop: 8,
  paddingBottom: 16,
  fontSize: 12,
  fontWeight: 400,
  color: '#7c7c7c',
  display: 'flex',
});

export const CommentReach = styled('div')({
  paddingRight: 16,
  paddingTop: 8,
  fontSize: 12,
  fontWeight: 400,
  color: '#7c7c7c',
  display: 'flex',
});

export const Video = styled('video')(({ src }) => ({
  src,
}));

export const Image = styled('img')(({ src }) => ({
  src,
}));

export const CommentContainer = styled(Link)({
  borderRadius: 4,
  backgroundColor: '#e9f5fd',
  fontSize: 14,
  margin: '0 12px 12px',
  padding: '4px 8px 8px',
  display: 'flex',
  fontWeight: 400,
  textAlign: 'left',
  color: 'black',
  flexDirection: 'row',
  textDecoration: 'none',
  '&:hover': {
    textDecoration: 'none',
  },
});

export const GoToThread = styled(Link)(() => ({
  display: 'block',
  fontSize: '12px',
  fontWeight: 400,
  margin: '0px 0px 4px 8px',
  minHeight: 0,
  textAlign: 'left',
  color: '#0079d3',
  paddingRight: 16,
  paddingLeft: 16,
  '&:hover': {
    textDecoration: 'none',
  },
}));
export const PostLink = styled(Link)(() => ({
  fontSize: '12px',
  fontWeight: 400,
  textAlign: 'left',
  color: 'black',
  textDecoration: 'none',
  '&:hover': {
    textDecoration: 'none',
  },
}));
