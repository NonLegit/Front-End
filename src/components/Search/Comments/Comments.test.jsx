import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Comments from './Comments';

// test snapshot
test('test snapshot', async () => {
  const props = {
    Comment:
      {
        _id: 'asd',
        author: {
          _id: 'asd212',
          userName: 'ahmed',
          profilePicture: 'https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/1610059_681891708529379_2132042836_n.jpg?_nc_cat=111&ccb=1-7&_nc_sid=174925&_nc_eui2=AeFisYcHx5f55wofAI89qsv4zwd8DyzyeI7PB3wPLPJ4jmThq4Z2nXImJdnzBzY682Ug8GJmq6pI9RMvTwe9n4Gn&_nc_ohc=njty5Km-D6IAX9S2nST&_nc_ht=scontent.fcai19-6.fna&oh=00_AfCuo4cKDBZx7mLXxK-A1PtFj7RKiSri443Fa8QU-Pb5yw&oe=63C8EB3B',
          displayName: 'milk',
        },
        post: {
          _id: 'asd212',
          author: {
            _id: 'asd212',
            userName: 'ahmed',
            profilePicture: 'https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2',
            displayName: 'hamada',
          },
          owner: {
            icon: 'https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2',
            _id: '12asda2',
            fixedName: 'Nonlegit',
            name: 'asdwd',
            membersCount: 123,
            description: 'this is my first subreddit',
          },
          title: 'first post',
          kind: 'image',
          text: 'asdasda',
          url: 'https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2',
          images: [
            'https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2',
            'https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2',
            'https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2',
          ],
          video: 'hamada with milk',
          votes: 1200,
          commentCount: 1300,

          createdAt: '2022-11-12T12:29:28Z',
          flairId: {
            _id: 'asd',
            text: 'asd',
            backgroundColor: 'blue',
            textColor: 'red',
          },
        },
        text: "aspidasjd;ioajf;sadfs;f adas daspdioasdklsa iodaskl;d jasdj sad iajfhdafhjsd fhad isd ijsa ja;sfh;jal;f h;jasdh uoas'jdash;jhas;fiasj das'jd asj dhuaos;d j;as'dk as jfa;sifj'las sa;dkl jnasjd as;f a;kf as d;asnfajlsf nla'f'klash fas'f ashf a'hf aio'f ha'fh 'ahfl ahsf hdajfjsdbgdalfja ;fdusghf;afh alndsg' hak fkDAJF ldanfl'kaf hal'kfh a'l hf'alsias  ihasdh asl'kfjash ahjfhasjf haslf hlas hfa'hslf haf'gaudlfNj'alsh 'g aspidasjd;ioajf;sadfs;f adas daspdioasdklsa iodaskl;d jasdj sad iajfhdafhjsd fhad isd ijsa ja;sfh;jal;f h;jasdh uoas'jdash;jhas;fiasj das'jd asj dhuaos;d j;as'dk as jfa;sifj'las sa;dkl jnasjd as;f a;kf as d;asnfajlsf nla'f'klash fas'f ashf a'hf aio'f ha'fh 'ahfl ahsf hdajfjsdbgdalfja ;fdusghf;afh alndsg' hak fkDAJF ldanfl'kaf hal'kfh a'l hf'alsias  ihasdh asl'kfjash ahjfhasjf haslf hlas hfa'hslf haf'gaudlfNj'alsh",
        createdAt: '2022-11-12T12:29:28Z',
        votes: 1200,
        repliesCount: 1222,
      },
  };
  const tree = renderer.create(

    <Router>
      <Comments {...props} />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
