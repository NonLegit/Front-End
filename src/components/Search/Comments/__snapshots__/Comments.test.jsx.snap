// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-1sesid4"
>
  <div
    className="MuiBox-root css-8atqhb"
  >
    <div
      className="MuiBox-root css-sxvjad"
    >
      <div
        className="MuiBox-root css-1lbt5xs"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular css-2is7ki-MuiAvatar-root"
        >
          <img
            alt="Profile Image"
            className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2"
          />
        </div>
        <a
          className="css-kqg5s"
          color="#000"
          fontWeight="bolder"
          href="/r/Nonlegit"
          onClick={[Function]}
        >
          r/
          asdwd
        </a>
        <div
          className="MuiBox-root css-1clzuz0"
        >
          <span>
            •
          </span>
          <div>
            Posted By
          </div>
          <div
            className="MuiAvatar-root MuiAvatar-circular css-2is7ki-MuiAvatar-root"
          >
            <img
              alt="Profile Image"
              className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/14731344_1268156939902850_4843578088361110846_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=09cbfe&_nc_eui2=AeEky4_3cTAH8AsHcH3UuXIKcLp-k-70xXVwun6T7vTFdcArTit-okcWlHaag26qfJeWLi2DDiYbw59pVU7sfWCM&_nc_ohc=A5SpL8aNUVEAX-YO51A&_nc_oc=AQniPbizNXpZ3SFnkYVS1CpwqCbtRPI2D8DeAHUhOG0t0JSuAtT4Bvl1Xbu4WrCWCn8&_nc_ht=scontent.fcai19-6.fna&oh=00_AfDhRAFvWuZtkXlwquZXICIDBSAhT2eF7HurP0EfQHnCEg&oe=63C6E9D2"
            />
          </div>
          <a
            className="css-1fa4cpj"
            color="inherit"
            fontWeight="normal"
            href="/user/ahmed"
            onClick={[Function]}
          >
            u/
            hamada
             
            a month ago
          </a>
        </div>
      </div>
    </div>
    <a
      className="css-1jla3h9"
      href="/"
      onClick={[Function]}
    >
      <a
        className="css-9d0c2z"
        href="/subreddit/Nonlegit/comments/asd212"
        onClick={[Function]}
      >
        <h3
          className="MuiTypography-root MuiTypography-h6 css-b4dk1a-MuiTypography-root"
        >
          first post
        </h3>
      </a>
      <a
        className="css-1h1xqtx"
        href="/subreddit/Nonlegit/comments/asd212/asd"
        onClick={[Function]}
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular css-1vmzm7i-MuiAvatar-root"
        >
          <img
            alt="Profile Image"
            className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://scontent.fcai19-6.fna.fbcdn.net/v/t1.18169-9/1610059_681891708529379_2132042836_n.jpg?_nc_cat=111&ccb=1-7&_nc_sid=174925&_nc_eui2=AeFisYcHx5f55wofAI89qsv4zwd8DyzyeI7PB3wPLPJ4jmThq4Z2nXImJdnzBzY682Ug8GJmq6pI9RMvTwe9n4Gn&_nc_ohc=njty5Km-D6IAX9S2nST&_nc_ht=scontent.fcai19-6.fna&oh=00_AfCuo4cKDBZx7mLXxK-A1PtFj7RKiSri443Fa8QU-Pb5yw&oe=63C8EB3B"
          />
        </div>
        <div
          className="MuiBox-root css-0"
        >
          <div
            className="MuiBox-root css-11kldd9"
          >
            <a
              className="css-kqg5s"
              color="#000"
              fontWeight="bolder"
              href="/user/ahmed"
              onClick={[Function]}
            >
              milk
            </a>
            <div
              className="MuiBox-root css-1clzuz0"
            >
              <span
                className="MuiTypography-root MuiTypography-body1 css-vcah1l-MuiTypography-root"
              >
                •
              </span>
              <a
                className="css-1fa4cpj"
                color="inherit"
                fontWeight="normal"
                href="/"
                onClick={[Function]}
              >
                a month ago
              </a>
            </div>
          </div>
          <h3
            className="MuiTypography-root MuiTypography-h6 css-1rxmjur-MuiTypography-root"
          >
            aspidasjd;ioajf;sadfs;f adas daspdioasdklsa iodaskl;d jasdj sad iajfhdafhjsd fhad isd ijsa ja;sfh;jal;f h;jasdh uoas'jdash;jhas;fiasj das'jd asj dhuaos;d j;as'dk as jfa;sifj'las sa;dkl jnasjd as;f a;kf as d;asnfajlsf nla'f'klash fas'f ashf a'hf aio'f ha'fh 'ahfl ahsf hdajfjsdbgdalfja ;fdusghf;afh alndsg' hak fkDAJF ldanfl'kaf hal'kfh a'l hf'alsias  ihasdh asl'kfjash ahjfhasjf haslf hlas hfa'hslf haf'gaudlfNj'alsh 'g aspidasjd;ioajf;sadfs;f adas daspdioasdklsa iodaskl;d jasdj sad iajfhdafhjsd fhad isd ijsa ja;sfh;jal;f h;jasdh uoas'jdash;jhas;fiasj das'jd asj dhuaos;d j;as'dk as jfa;sifj'las sa;dkl jnasjd as;f a;kf as d;asnfajlsf nla'f'klash fas'f ashf a'hf aio'f ha'fh 'ahfl ahsf hdajfjsdbgdalfja ;fdusghf;afh alndsg' hak fkDAJF ldanfl'kaf hal'kfh a'l hf'alsias  ihasdh asl'kfjash ahjfhasjf haslf hlas hfa'hslf haf'gaudlfNj'alsh
          </h3>
          <div
            className="css-fhwtbh"
          >
            <span
              className="MuiTypography-root MuiTypography-span css-rjslxy-MuiTypography-root"
            >
              1.2K
               
              upvotes
               
            </span>
            <span
              className="MuiTypography-root MuiTypography-span css-rjslxy-MuiTypography-root"
            >
               
              awards
               
            </span>
          </div>
        </div>
      </a>
    </a>
    <a
      className="css-pbvgqa"
      href="/subreddit/Nonlegit/comments/asd212/asd"
      onClick={[Function]}
    >
      Go to thread
    </a>
    <div
      className="css-1wx6jdt"
    >
      <span
        className="MuiTypography-root MuiTypography-span css-1askrqg-MuiTypography-root"
      >
        1.2K
         
        upvotes
         
      </span>
      <span
        className="MuiTypography-root MuiTypography-span css-1askrqg-MuiTypography-root"
      >
        1.3K
         
        comments
         
      </span>
    </div>
  </div>
</div>
`;
